### PR TITLE
dirichlet extension

### PR DESCRIPTION
**Pull request by nassimberrada**
Throws the following error for all back-ends except numpy: 

`>       ret, ret_gt = call()`
`E       TypeError: cannot unpack non-iterable NoneType object`

### CODE REVIEW

File: ivy/array/extensions/random.py [hunk 1]
```diff
@@ -1,6 +1,61 @@
...
+
+        Returns
+        -------
+        ret
+            The drawn samples, of shape (size, k).
+
+        Examples
+        --------
+        >>> alpha = ivy.array([1.0, 2.0, 3.0])
+        >>> ivy.dirichlet(alpha)
```
Comment (Ishticode) [inline ivy/array/extensions/random.py:48]: Should be an instance method... and below too

**Review by Ishticode — APPROVED** [review]
I have fixed tests for all backends....
Let me know if you have any objection to the updates.
There was several errors. Could explain if required.
But the one you mentioned above was caused because the `call()` upon `test_funnction` was hitting a `return` (equivalent to skip test) which return None and hence it was unable to unpack it `ret` and `ret_gt`.

Anyway feel free to merge after taking care of the below comment and the lint (try linking up the issue if possible) ...

**Comment by nassimberrada** [comment]
So the issue was with the testing, i'm still wondering why the tests passed with numpy if that's the case 🤔. In any case, glad this one is finally done, been dragging it around since the beginning of my work at Ivy hahah, thanks !

**Comment by nassimberrada** [comment]
Close #6156

**Pull request merged**